Lib/fontTools/ttLib/ttFont.py

```diff
@@ -101,6 +102,7 @@ def __init__(self, file=None, res_name_or_index=None,
 		self.recalcTimestamp = recalcTimestamp
 		self.tables = {}
 		self.reader = None
+		self.cfg = Config(cfg)
```

Comment:
how about here you check `if not isinstance(cfg, AbstractConfig): cfg = Config(cfg)` otherwise you self `self.cfg = cfg`. This would allow one to stick a custom super-set of fonttools own Config with additional flags that can be useful in ones workflow

Reply:
Ok will do now

Reply:
We would still want each TTFont instance to have its distinct copy, since a config is mutable.
You current cast `Config(cfg)` achieves that already. To accomplish the same with my proposal above we need a `AbstractConfig.copy()` method that returns a new instance of the type with its own copy of the internal `self._values` dictionary. A shallow copy is fine, if any mutable values are used in individual configs, it's not our job to deepcopy those I think.

So in addition to [my earlier suggestion](https://github.com/fonttools/fonttools/pull/2416/files#r850336161), we add this to `AbstractConfig`:

```python
def copy(self):
    return self.__class__(self._values)
```

and then call it inside `TTFont.__init__`

```python
    self.cfg = Config(cfg) if not isinstance(cfg, AbstractConfig) else cfg.copy()
```

Reply:
maybe also add a test that sets identical Config oject in two different fonts, and assert they have different `id()`;  or modify one font.cfg, and assert that it did not also change in the other font

Reply:
Done, I added a test for the use-case that you were describing, if you can check that it's what you meant?

Reply:
> if you can check that it's what you meant?

yes, that's it

Reply:
Done for real now including the copy()